### DISCUSSION TITLE
How to bundle workspace dependencies into `pnpm publish` packed packages?

### DISCUSSION THREAD

**Discussion by douglasg14b** [pnpm/pnpm#8565] [Q&A] [answered]
When we publish a package to a repository, workspace (`"@myworksapce/some-package": "workspace:*"`) dependencies don't seem to be bundled with, and remain as `package.json` dependencies (albeit, with an actual version replacing the workspace protocol). 

This prevents the published package from being installed as these packages are not published to a package repository, and cannot be pulled from a package manager.

How do we get workspace package dependencies to bundle into published packages without it prompting package install to install private workspace packages?

The docs here https://pnpm.io/workspaces#publishing-workspace-packages describe the behavior, but doesn't necessarily describe how to bundle workspace-only dependencies into published packages.

**Comment by douglasg14b** [comment, marked as answer]
This can be done with `bundleDependencies` https://docs.npmjs.com/cli/v10/configuring-npm/package-json#bundledependencies

**Reply by PabloStockhausen** [reply]
@douglasg14b, do you happen to have a reference project that demonstrates the bundling process? Did you utilize `pnpm publish` with just the `bundleDependencies` option for this, or did you opt for `pnpm deploy` or `pnpm pack` beforehand? 

I am running into the exact same issue and can't resolve it with just adding `bundleDependencies`.

Thanks in advance for any insights!
